[docs] Explain the information listed on the system properties page (#23566)
mui-org/material-ui: docs/src/pages/system/properties/properties.md

# Properties

<p class="description">This page lists all of the custom system properties, how are they linked with the theme, and which CSS properties they compute.</p>

> Let's take the following row, for example:

Reviewer: ```suggestion
Let's take the following row from [the table below](#properties-reference-table), for example:
```
?

| Group                       | System style function | System key(s)        | CSS property/properties | Theme mapping                                                                 |
| :-------------------------- | :-------------------- | :------------------- | :---------------------- | :---------------------------------------------------------------------------- |
| [spacing](/system/spacing/) | `spacing`             | `mb`, `marginBottom` | `margin-bottom`         | [`theme.spacing(value)`](/customization/default-theme/?expand-path=$.spacing) |

1. The <b>Group</b> column links to the documentation page where this group of properties is described; in this example, the [spacing](/system/spacing/) page.

> 2. The <b>System style function</b> references the function which generates the properties shown in the other columns, in case you want to use it for adding this functionality to your custom components. The function can be imported from `@material-ui/system`.

Reviewer: ```suggestion
2. The <b>System style function</b> column references the function which generates the properties shown in the other columns, in case you want to use it for adding this functionality to your custom components. The function can be imported from `@material-ui/system`.
```

```jsx
import * as React from 'react';
import styled from 'styled-components';
import { spacing } from '@material-ui/system';

const Div = styled('div')(spacing);

export default function Demo() {
  return <Div mb={3} />;
}
```

3. The <b>System keys</b> column lists the key(s) by which you can use this property with the `sx` prop (or as a system function).
4. The <b>CSS property</b> column describes which CSS property will be generated when this system property is used.

> 5. And lastly, the <b>Theme key</b> column tells you how this property is wired with the theme - with this example, whatever value you might use will be used as input to the `theme.spacing` helper.

Reviewer: ```suggestion
5. And lastly, the <b>Theme key</b> column tells you how this property is wired with the theme – with this example, whatever value you might use will be used as input to the `theme.spacing` helper.
```

> Let's take a look at one example:

Reviewer: ```suggestion
Let's take a look at an example:
```

```jsx
<Box sx={{ mb: 3 }} />

> // equivalent as

Reviewer: ```suggestion
// is equivalent to
```

<Box sx={{ marginBottom: theme => theme.spacing(3)}} />
```

> As the default theme spacing is 8px, this will result in the following CSS class

Reviewer: ```suggestion
As the default theme spacing is 8px, this will result in the following CSS class:
```

```css
.hash-MuiBox {
  margin-bottom: 24px;
}
```

## Properties reference table

> | Group                             | System style function | System key         | CSS property/properties                                                                      | Theme mapping                                                                       |

Reviewer: ```suggestion
| Group                             | System style function | System key(s)         | CSS property/properties                                                                      | Theme mapping                                                                       |
```

| [sizing](/system/sizing/)         | `minHeight`           | `minHeight`        | `min-height`                                                                                 | none                                                                                |
| [sizing](/system/sizing/)         | `minWidth`            | `minWidth`         | `min-width`                                                                                  | none                                                                                |
| [sizing](/system/sizing/)         | `width`               | `width`            | `width`                                                                                      | none                                                                                |
| [sizing](/system/sizing/)         | `boxSizing`           | `boxSizing`        | `box-sizing`                                                                                 | none                                                                                |

> | [spacing](/system/spacing/)       | `spacing`             | `m,margin`         | `margin`                                                                                     | [`theme.spacing(value)`](/customization/default-theme/?expand-path=$.spacing)       |

Reviewer: ```suggestion
| [spacing](/system/spacing/)       | `spacing`             | `m`, `margin`         | `margin`                                                                                     | [`theme.spacing(value)`](/customization/default-theme/?expand-path=$.spacing)       |
```
(etc.)

# Properties

> <p class="description">This page lists all custom system properties, how are they linked with the theme, and which CSS properties are they resulting in.</p>

Reviewer 2: The table doesn't suggest to me "how" the system properties are linked to the them. Just to which one. Do they use that value for computation?

Author: Let me try to refactor that colum. Maybe instead of just `spacing` we can use a short code snippet like:
`theme.spacing(inputValue)`. What do you think?

Author: Done with https://github.com/mui-org/material-ui/pull/23566/commits/50271358c1f3653bfd89cae83439644c3f26cbe4 Let me know what you think

> <p class="description">This page lists all custom system properties, how are they linked with the theme, and which CSS properties are they resulting in.</p>

Reviewer 2: ```suggestion
<p class="description">This page lists all custom system properties, how are they linked with the theme, and which CSS properties they compute.</p>
```

Need to read the sentence over and over. Didn't "feel" like correct english to me.

> For example, let's the the following row:

Reviewer 3: ```suggestion
For example, let's take the following row:
```

> For example, let's the the following row:

Reviewer 2: ```suggestion
For example, let's check the following row:
```
? Or some other verb

Author: `take` was my intention :)

Reviewer 2: Works as well. Just used the `suggestion` to show exactly where the issue is.

| Group                       | System style function | System key        | CSS property/properties | Theme key                                                        |
| :-------------------------- | :-------------------- | :---------------- | :---------------------- | :--------------------------------------------------------------- |
| [spacing](/system/spacing/) | `spacing`             | `mb,marginBottom` | `margin-bottom`         | [`spacing`](/customization/default-theme/?expand-path=$.spacing) |

> 1. The <b>Group</b> column gives you access to the documentation page where this group of properties are described, in the example, the [spacing](/system/spacing/) page.

Reviewer 3: ```suggestion
1. The <b>Group</b> column gives you access to the documentation page where this group of properties is described, in the example, the [spacing](/system/spacing/) page.
```

> 1. The <b>Group</b> column gives you access to the documentation page where this group of properties are described, in the example, the [spacing](/system/spacing/) page.

Reviewer 2: ```suggestion
1. The <b>Group</b> column gives you access to the documentation page where this group of properties is described, in the example, the [spacing](/system/spacing/) page.
```

> 2. The <b>System style function</b> is referencing the style function which generates these properties, in case you want to use it for adding this functionality to your custom components. The function can be imported from `@material-ui/system`.

Reviewer 2: ```suggestion
2. The <b>System style function</b> is referencing the style function which generates the properties in the other columns, in case you want to use it for adding this functionality to your custom components. The function can be imported from `@material-ui/system`.
```

> 2. The <b>System style function</b> is referencing the style function which generates these properties, in case you want to use it for adding this functionality to your custom components. The function can be imported from `@material-ui/system`.

Reviewer 2: Can we link to an example here how one would use it?

Author: Done with https://github.com/mui-org/material-ui/pull/23566/commits/50271358c1f3653bfd89cae83439644c3f26cbe4

Reviewer 2: Shouldn't we have a dedicated page for that and not just some inlined example? How would I find this if I just want to use `@material-ui/system/properties` and not `sx` properties?

Reviewer 3: We have started to cover this topic during Friday's meeting. There are a couple of aspects of the system that we do no longer document. These aspects could potentially lead to breaking changes or new documentation pages. From my perspective, there are two aspects:

1. Cherry-picking some sub-parts of the system. This is what this part of the documentation cover. Do we even need this in the first place?
2. Extensibility, how to had custom utils into the system?

In any case, I think that it goes beyond the scope of the pull request, but is definitely important to discuss to get a solid next step.

Author: We talked about adding Advanced page where we would dive into details of how the style function can be used. I believe that one could give more details. Should I add it as part of this PR so we can have the whole system updated, or would it be ok if I open new PR for that?

Reviewer 2: If we export the system utilities we need to document them. Having this documentation living on an unrelated page is not really user-friendly.

Reviewer 3: > If we export the system utilities we need to document them

We have two options, remove the exports or add the documentation. I think that what's best for v5 should influence the outcome, not the legacy.

> Having this documentation living on an unrelated page is not really user-friendly.

Yeah, I think that we could add a link to a dedicated page with more details.

Author: @eps1lon  I agree! @oliviertassinari according to our conversation, if we don't plan to advertise this functionality, we may drop this column. If not, I can open next PR for advanced usage and link these column to appropriate section there. I am ok with both options, leaning towards dropping the column as the `experimentalStyled` utility already supports adding the `sx` prop functionality on custom component. Adding additional style functions is already available by the `styled()` API so I am not even sure if this will really be used that much...

Author: Alright, will add as next PR the advanced page and will update this section. Not sure how much those will be used but it will be interesting to see anyway

Reviewer 2: Sounds good. Leaving this conversation as unresolved so that we don't forget.

3. The <b>System key</b> column, gives you the key/keys by which you may use this property as part of the system (or the `sx` prop).

> 4. The <b>CSS proprety</b> column describes which CSS property will be generated when this system prop is used.

Reviewer 2: ```suggestion
4. The <b>CSS property</b> column describes which CSS property will be generated when this system property is used.
```

5. And lastly, the <b>Theme key</b> column tells you how this property is wired with the theme - with this property, whatever value you may use will be used as input to the `theme.spacing` helper.

> Let's take a look on one example:

Reviewer 2: ```suggestion
Let's take a look at one example:
```
? /cc @mbrookes

```jsx
<Box sx={{ mb: 3 }} />

// equivalent as
<Box sx={{ marginBottom: theme => theme.spacing(3)}} />
```

> As the default theme spacing is 8, this will result in the following CSS class

Reviewer 2: ```suggestion
As the default theme spacing is 8px, this will result in the following CSS class
```
